Scripts/Entities/Commands/EntityCommandMovement.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Sankari;
+
+public interface IEntityMovement : IEntityMoveable, IEntityDash
+{
+	// Ground acceleration
+	public int GroundAcceleration { get; set; }
+
+}
+
+public class EntityCommandMovement : EntityCommand<IEntityMovement>
+{
+	public int MaxSpeedWalk     { get; set; } = 350;
+	public int MaxSpeedSprint   { get; set; } = 500;
+	public int MaxSpeedAir      { get; set; } = 350;
+	public int AirAcceleration  { get; set; } = 20;
+	public int DampeningAir     { get; set; } = 10;
+	public int DampeningGround  { get; set; } = 25;
+
+	public EntityCommandMovement(IEntityMovement entity) : base(entity) { }
+
+	public override void Initialize()
+	{
+		// if these are equal to each other then the player movement will not work as expected
+		if (Entity.GroundAcceleration == DampeningGround)
+			DampeningGround -= 1;
+	}
+
+	public override void UpdateGroundWalking(float delta)
+	{
+		var velocity = Entity.Velocity;
+		velocity.x = ClampAndDampen(velocity.x, DampeningGround, MaxSpeedWalk);
+		Entity.Velocity = velocity;
+	}
+
+	public override void UpdateGroundSprinting(float delta)
+	{
+		var velocity = Entity.Velocity;
+		velocity.x = ClampAndDampen(velocity.x, DampeningGround, MaxSpeedSprint);
+		Entity.Velocity = velocity;
```

Comment:
Optional:
```suggestion
		Velocity = new Vector2(ClampAndDampen(velocity.x, DampeningGround, MaxSpeedSprint),Velocity.y);
```

For these and other similar functions as well. Update air is probably fine.

Reply:
![image](https://user-images.githubusercontent.com/6277739/195206934-0edaf9cb-83b0-45b1-9ca8-cae519c742fa.png)

I prefer `B`